docs/sources/reference/components/otelcol/otelcol.processor.probabilistic_sampler.md

```diff
@@ -90,6 +114,279 @@ Name    | Type               | Description
+Users can randomness values in this way, independently, making it
+possible to apply consistent sampling across traces for example.  If
+the Trace was initialized with pre-determined randomness value
+`9b8233f7e3a151` and 100% sampling, it would read:
+
+```
+tracestate: ot=th:0;rv:9b8233f7e3a151
+```
+
+This component, using either proportional or equalizing modes, could
+apply 50% sampling the Span.  This span with randomness value
+`9b8233f7e3a151` is consistently sampled at 50% because the threshold,
+when zero padded (i.e., `80000000000000`), is less than the randomess
+value.  The resulting span will have the following tracestate:
+
+```
+tracestate: ot=th:8;rv:9b8233f7e3a151
+```
+
+For log records, threshold and randomness information are encoded in
+the log record itself, using attributes.  For example, 25% sampling
+with an explicit randomness value is encoded as:
+
+```
+sampling.threshold: c
+sampling.randomness: e05a99c8df8d32
+```
+
+### Sampling precision
+
+When encoding sampling probability in the form of a threshold,
+variable precision is permitted making it possible for the user to
+restrict sampling probabilities to rounded numbers of fixed width.
+
+Because the threshold is encoded using hexadecimal digits, each digit
+contributes 4 bits of information.  One digit of sampling precision
+can express exact sampling probabilities 1/16, 2/16, ... through
+16/16.  Two digits of sampling precision can express exact sampling
+probabilities 1/256, 2/256, ... through 256/256.  With N digits of
+sampling precision, there are exactly `(2^N)-1` exactly representable
+probabilities.
+
+Depending on the mode, there are different maximum reasonable settings
+for this parameter.
+
+- The `hash_seed` mode uses a 14-bit hash function, therefore
+  precision 4 completely captures the available information.
+- The `equalizing` mode configures a sampling probability after
+  parsing a `float32` value, which contains 20 bits of precision,
+  therefore precision 5 completely captures the available information.
+- The `proportional` mode configures its ratio using a `float32`
+  value, however it carries out the arithmetic using 56-bits of
+  precision.  In this mode, increasing precision has the effect
+  of preserving precision applied by preceding samplers.
+
+In cases where larger precision is configured than is actually
+available, the added precision has no effect because trailing zeros
+are eliminated by the encoding.
+
+### Error handling
+
+This processor considers it an error when the arriving data has no
+randomness.  This includes conditions where the TraceID field is
+invalid (16 zero bytes) and where the log record attribute source has
+zero bytes of information.
+
+By default, when there are errors determining sampling-related
+information from an item of telemetry, the data will be refused.  This
+behavior can be changed by setting the `fail_closed` property to
+false, in which case erroneous data will pass through the processor.
+
```

Comment:
We try to avoid that, since it contains things that are specific to the upstream Collector and it may confuse users.

There are other docs where we do link to a lot of upstream docs. E.g. we do it in the [otelcol.processor.transform](https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.processor.transform/) docs. But there we link to OTTL documentation which I believe almost entirely also applies to Alloy, and so the likelihood that users would be confused is low.

Reply:
I agree we should avoid verbatim copy/pasting from an external source... 

I would only sign off on this if:
1. The source content is open source documentation and the licence allows reuse (looks like this is under Apache 2.0)
1. We give clear attribution per the license it's released under

Otherwise we need to link out to the reference material. We can be explicit in the linking info what is applicable and what is beyond the scope of Alloy if there's a chance of confusion.

Reply:
We also risk being out of sync with the original text: with such a big wall of text, it's easy to miss that something changed during the very annoying process of updating the otel dependencies

Reply:
Ok, sure! I replaced this with a link to the upstream docs.